src/pathTemplate.ts

```diff
@@ -118,69 +99,134 @@ export class PathTemplate {
    *   parsed
    */
   render(bindings: Bindings): string {
-    const out: Segment[] = [];
-    let inABinding = false;
-    this.segments.forEach(segment => {
-      if (segment.kind === extras.BINDING) {
-        if (!has(bindings, segment.literal)) {
-          const msg = util.format(
-            'Value for key %s is not provided in %s',
-            segment.literal,
-            bindings
-          );
-          throw new TypeError(msg);
-        }
-        const tmp = new PathTemplate(bindings[segment.literal].toString());
-        Array.prototype.push.apply(out, tmp.segments);
-        inABinding = true;
-      } else if (segment.kind === extras.END_BINDING) {
-        inABinding = false;
-      } else if (inABinding) {
-        return;
-      } else {
-        out.push(segment);
+    if (Object.keys(bindings).length !== Object.keys(this.bindings).length) {
+      throw new TypeError('The number of variables does not match');
```

Comment:
This error message is not actionable. The previous one, `'Value for key %s is not provided in %s'`, gives some clue about what's wrong. Can we improve it here?

Please note that all the errors thrown from here will be user visible. Expect to receive a client library issue for all the error messages that are just a little bit unclear :)

Reply:
sure, here it will complain if the given bindings do not have the same variable number with the needed variables.
This sounds better maybe:
```
        `The number of variables ${
          Object.keys(bindings).length
        } does not match the number of needed variables ${
          Object.keys(this.bindings).length
        }`
```